### PR TITLE
Support NEW and DUP

### PR DESCRIPTION
**Pull request by saberduck**

### CODE REVIEW

File: java-frontend/src/test/java/org/sonar/java/bytecode/se/BytecodeEGWalkerExecuteTest.java [hunk 1]
```diff
@@ -20,6 +20,7 @@
 package org.sonar.java.bytecode.se;
 
 import org.junit.Test;
+import org.mockito.internal.matchers.Null;
```
Comment (benzonico) [inline java-frontend/src/test/java/org/sonar/java/bytecode/se/BytecodeEGWalkerExecuteTest.java]: useless import.

File: java-frontend/src/test/java/org/sonar/java/bytecode/cfg/BytecodeCFGConstructionTest.java [hunk 2]
```diff
@@ -188,12 +194,18 @@ public BytecodeCFGConstructionTest(TestInput testInput, BytecodeCFGBuilder.Instr
...
   public void test() throws Exception {
-    int opcode = testInput.opcode;
-    if (isJumpInstruction(opcode)) {
+    if (isJumpInstruction(testInput.opcode)) {
       test_jumps();
       return;
     }
-    BytecodeCFGBuilder.BytecodeCFG cfg = new Instructions().cfg(testInput.opcode, testInput.operandOrVar);
+    BytecodeCFGBuilder.BytecodeCFG cfg;
+    if (Instructions.TYPE_INSN.contains(testInput.opcode)) {
```
Comment (benzonico) [inline java-frontend/src/test/java/org/sonar/java/bytecode/cfg/BytecodeCFGConstructionTest.java]: This if should be part of the Instructions class. not of this test.

File: java-frontend/src/main/java/org/sonar/java/bytecode/se/BytecodeEGWalker.java [hunk 3]
```diff
@@ -152,10 +152,18 @@ void executeInstruction(BytecodeCFGBuilder.Instruction instruction) {
       case Opcodes.SALOAD:
         break;
       case Opcodes.LDC:
+      case Opcodes.NEW: {
         SymbolicValue symbolicValue = constraintManager.createSymbolicValue(instruction);
         programState = programState.stackValue(symbolicValue);
         programState = programState.addConstraint(symbolicValue, ObjectConstraint.NOT_NULL);
         break;
+      }
+      case Opcodes.DUP: {
```
Comment (SonarTech) [inline java-frontend/src/main/java/org/sonar/java/bytecode/se/BytecodeEGWalker.java]: MINOR Extract this nested code block into a method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1199)

File: java-frontend/src/main/java/org/sonar/java/bytecode/se/BytecodeEGWalker.java [hunk 4]
```diff
@@ -152,10 +152,18 @@ void executeInstruction(BytecodeCFGBuilder.Instruction instruction) {
       case Opcodes.SALOAD:
         break;
       case Opcodes.LDC:
+      case Opcodes.NEW: {
```
Comment (SonarTech) [inline java-frontend/src/main/java/org/sonar/java/bytecode/se/BytecodeEGWalker.java]: MINOR Extract this nested code block into a method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1199)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**